ground/src/main/java/com/google/android/ground/persistence/local/room/entity/PhotoMutationEntity.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.persistence.local.room.entity
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Index
+import androidx.room.PrimaryKey
+import com.google.android.ground.persistence.local.room.fields.MutationEntitySyncStatus
+import com.google.android.ground.persistence.local.room.fields.MutationEntityType
+
+/**
+ * Persistent database representation of a [com.google.android.ground.model.mutation.PhotoMutation].
+ */
+@Entity(
+  tableName = "photo_mutation",
+  foreignKeys =
+    [
+      ForeignKey(
+        entity = SubmissionMutationEntity::class,
+        parentColumns = ["id"],
+        childColumns = ["submission_mutation_id"],
+        onDelete = ForeignKey.CASCADE
+      ),
+    ],
+  indices = [Index("submission_mutation_id")]
+)
+data class PhotoMutationEntity(
```

Comment:
If we can refactor the submission mutation logic as mentioned above we can avoid a lot of complexity - the new DAO, entity, mutation, conversion logic, etc.